### PR TITLE
Feature/kabo5 nrl 511 capability statements

### PR DESCRIPTION
**Pull request by katebobyn-nhs**

### CODE REVIEW

File: terraform/infrastructure/producer.tftpl [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            "referencePolicy": ["logical"],
+            "searchParam": [
+              {
+                "name": "type",
+                "definition": "http://hl7.org/fhir/SearchParameter/DocumentReference-type",
+                "type": "token",
+                "documentation": "Allows DocumentReference search results to be filtered by pointer type (SNOMED code)."
+              },
+              {
+                "name": "subject",
```
Comment (Tomdango) [inline terraform/infrastructure/producer.tftpl:97]: Hmm... should the name of this be `subject:identifier`? I think we should raise a ticket to rename `subject:identifier` to just `subject` and then correct this.

**Review by Tomdango — COMMENTED** [review]
Brain dump review 🧠

File: terraform/infrastructure/modules/api_gateway/api_gateway.tf [hunk 2]
```diff
@@ -9,12 +9,54 @@ resource "aws_api_gateway_rest_api" "api_gateway_rest_api" {
   }))
 }
 
+resource "aws_api_gateway_resource" "capability" {
+  rest_api_id = aws_api_gateway_rest_api.api_gateway_rest_api.id
+  parent_id   = aws_api_gateway_rest_api.api_gateway_rest_api.root_resource_id
+  path_part   = "_metadata"
```
Comment (Tomdango) [inline terraform/infrastructure/modules/api_gateway/api_gateway.tf:15]: metadata

File: terraform/infrastructure/producer.tftpl [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            "interaction": [
+              {
+                "code": "read"
+              }
+            ],
+            "versioning": "no-version"
+          },
+          {
+            "type": "DocumentReference",
+            "profile": "hl7.org/fhir/R4/documentreference.html",
```
Comment (katebobyn-nhs) [inline terraform/infrastructure/producer.tftpl:61]: http

File: terraform/infrastructure/producer.tftpl [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                "name": "type",
+                "definition": "http://hl7.org/fhir/SearchParameter/DocumentReference-type",
+                "type": "token",
+                "documentation": "Allows DocumentReference search results to be filtered by pointer type (SNOMED code)."
+              },
+              {
+                "name": "subject",
+                "definition": "http://hl7.org/fhir/SearchParameter/DocumentReference-subject",
+                "type": "reference",
+                "documentation": "Allows DocumentReference search results to be filtered by patient (NHS number)."
```
Comment (Tomdango) [inline terraform/infrastructure/producer.tftpl:107]: Add comment here that only `subject:identifier` is supported.

**Review by Tomdango — COMMENTED** [review]
Approved pending those two comments

**Review by Tomdango — APPROVED** [review]

**Pull request merged**